src/ray/gcs/test/redis_gcs_client_test.cc

```diff
@@ -306,7 +306,7 @@ class TaskTableTestHelper {
                                        std::shared_ptr<gcs::RedisGcsClient> client) {
     // Add a table entry.
     const auto task_id = RandomTaskId();
-    const int num_modifications = 3;
+    enum { num_modifications = 3 };
```

Comment:
hmmm this seems weird, is this a known msvc issue?

Reply:
I haven't looked it up, but it's a silly issue. Visual C++ sees it as a `const` variable that isn't captured, so it produces an error when it's used in the lambda. And if you capture it, then you get a signed/unsigned comparison warning. Clang sees it as a compile-time constant and doesn't require capture (in fact it produces a warning which we translate to an error if we try to capture). It might have to do with the move to `constexpr` in later C++ versions, I'm not sure. But in any case it's trivial to fix by just making it a compile-time constant via `enum`.

Reply:
seems like we should just capture it and fix the sign/unsigned comparison by changing `const int` to `const uint64_t`, which is what `CreateTaskTableData` in few lines below

Reply:
I don't think there's any benefit to that though, is there? I feel like it'd only hurt performance (and readability) to capture too many variables without any a benefit -- I think the compiler will do everything correctly and efficiently when it can see it's a constant?

Reply:
I think using `enum` in place of `const int` hurt readability. `enum` is for enumeration (and after C++11 we should use class enum). 

In addition this is inside a unit test, we don't need to worry about performance of capturing an integer.